exchanges/btcmarkets/btcmarkets_wrapper.go

```diff
@@ -55,6 +56,16 @@ func (b *BTCMarkets) SetDefaults() {
 	b.API.CredentialsValidator.RequiresBase64DecodeSecret = true
 	b.API.Endpoints.URLDefault = btcMarketsAPIURL
 	b.API.Endpoints.URL = b.API.Endpoints.URLDefault
+	// Specific authentication issues
+	b.ValidateCredentialError = func(err error) error {
```

Comment:
I like the concept. However, I think it can be simplified. You are calling this func: `ValidateCredentialError `  from `CheckTransientError` but these aren't really the same type of errors as whats being checked in there. I think you could just append these to your checks below and lower complexity eg
```
func (b *BTCMarkets) ValidateCredentials() error {
	acc, err := b.GetAccountInfo()
	if err != nil {
		transErr := b.CheckTransientError(err)
                if transErr != nil {
                       return transErr
                }
                 if strings.Contains(err.Error(), "InvalidAPIKey") ||
			strings.Contains(err.Error(), "InvalidAuthTimestamp") ||
			strings.Contains(err.Error(), "InvalidAuthSignature") ||
			strings.Contains(err.Error(), "InsufficientAPIPermission") {
			return err
		}
	}
      
	return account.Process(&acc)
}
```

Reply:
Yeah maybe I was over thinking it. I just wanted to make sure that there was something that is specific to an exchange that was obvious to a dev. But will change to your way.